lib/ansible/module_utils/facts.py

```diff
@@ -412,6 +413,12 @@ def get_distribution_facts(self):
                                 if release:
                                     self.facts['distribution_release'] = release.groups()[0]
                                 break
+                            else:
+                                try:
+                                    self.get_lsb_facts()
+                                    self.facts['distribution_release'] = self.facts['lsb']['codename']
+                                except:
+                                    continue
                         elif name == 'Mandriva':
```

Comment:
If I read this right I think we're relying on get_platfrm() for everything except distribution_release?  If so we probably want the first break (just above your new code) to be in the if-block.  that way we exit once facts['distribution_release'] is successfully found.

If we're only scanning one file for the distribution_release then we probably need to remove the else block and instead have a condition that checks that we're actually looking at a debian system now (from self.facts['distribution'] perhaps?)  if we are not, continue.  If we are, then try the lsb facts to get the distribution_release.

If that's successful break.

If it's unsuccessful, we may break or fail... that depends on whether we're processing one or multiple files to determine the Debian distribution_release.